core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -5652,6 +5690,22 @@ components:
           role:
             description: the role to modify for the specified member
             $ref: '#/components/schemas/ProjectRole'
+    BatchProjectAccessUpdate:
+        required:
+          - membersToAdd
+          - membersToRemove
+        type: object
+        properties:
+          membersToAdd:
+            description: The list of members to add to the billing project
+            type: array
+            items:
+              $ref: '#/components/schemas/ProjectAccessUpdate'
+          membersToRemove:
+            description: The list of members to remove from the billing project
+            type: array
+            items:
+              $ref: '#/components/schemas/ProjectAccessUpdate'
```

Comment:
Does this mean consumers of this API need to provide a role when removing users from a project? Would it make sense to have this be an array of emails?

Reply:
The reason for implementing it this way is that a user can have multiple roles on the billing project, so this would allow the caller to revoke a single role. That said, I can't think of a good use-case for a user to have multiple roles so I could buy the argument that this is just an implementation detail of Sam that's leaking out and there's no strong reason to present it to the end-user this way.

I'm not tied to this current implementation, happy to make your suggested change if you'd like

Reply:
I took a look at the current [delete billing project member endpoint](https://github.com/broadinstitute/rawls/blob/5fb91e5e39cbc35c3a813dcfdda9c27dc45a119e/core/src/main/resources/swagger/api-docs.yaml#L592) and it accepts only an email. So, I'd say this new endpoint should consume only emails instead of email+role tuples, to keep it consistent.

Reply:
That endpoint also requires the role name to be specified as part of the [path](https://github.com/broadinstitute/rawls/blob/5fb91e5e39cbc35c3a813dcfdda9c27dc45a119e/core/src/main/resources/swagger/api-docs.yaml#L591). So leaving it as-is would be the more consistent approach